src/imaps/base/sheet.py

```diff
@@ -0,0 +1,316 @@
+                        ascii_value.strip()
+                        ascii_row_content.append(ascii_value)
+                content.append(ascii_row_content)
+
+        if self.extension == "tab.gz":
+            with gzip.open(self.fname, "rt") as unzipped_annotation:
+                annotation_content = list(csv.reader(unzipped_annotation, delimiter="\t"))
+                column_names = annotation_content[0]
+                content = annotation_content[1:]
+
+        self.content = content
+        self.column_names = column_names
+
+    def write_tab_file(self, path=None):
+        """Write a compressed tab delimited file."""
+        if path is None:
+            path = self.annotation_tab_name
+
+        with gzip.open(path, "wt") as tab_file:
+            tab_writer = csv.writer(tab_file, delimiter="\t")
+            tab_writer.writerow(self.column_names)
+            tab_writer.writerows(self.content)
+
+    def get_extension(self):
+        """Obtain the full extension."""
+        extension = self.fname.split(".")[-1]
+
+        if extension == "gz":
+            extension = ".".join(self.fname.split(".")[-2:])
+
+        self.extension = extension
+
+    def get_column(self, column_name):
+        """Obtain named column content."""
+        column_index = next(i for i, name in enumerate(self.column_names) if name == column_name)
+        return [row[column_index] for row in self.content]
+
+    def get_element(self, column_name, sample_name):
+        """Obtain content of specific sample in the named column."""
+        if column_name not in self.column_names:
+            raise ValueError(f'There is no column with column name: "{column_name}"')
+        if sample_name not in self.sample_names:
+            raise ValueError(f'There is no sample with sample name: "{sample_name}"')
+
+        for row, sample in zip(self.content, self.sample_names):
+            for element, column in zip(row, self.column_names):
+                if sample == sample_name and column == column_name:
+                    return element
+
+    def get_barcodes3(self):
+        """Get 3' barcodes and number of tolerated mismatches."""
+        barcodes3 = []
+        mismatches = []
+        for barcode in self.get_column("Linker"):
+            barcodes3.append(re.sub(r"[^ACGTN]+", "", barcode))
+            mismatches.append(int(re.sub(r"[^\d]+", "", barcode)))
+
+        assert len(set(mismatches)) == 1
+        return barcodes3, mismatches[0]
+
+    def get_barcodes5(self):
+        """Get 5' barcodes and number of tolerated mismatches."""
+        barcodes5 = []
+        mismatches = []
+        for barcode in self.get_column("5' barcode"):
+            barcodes5.append(re.sub(r"[^ACGTN]+", "", barcode))
+            mismatches.append(int(re.sub(r"[^\d]+", "", barcode)))
+        assert len(set(mismatches)) == 1
+        return barcodes5, mismatches[0]
+
+    def get_adapter(self):
```

Comment:
This method should be names `get_adapters` and the docstring should be changed accordingly.